waltid-libraries/waltid-library-commons/src/commonMain/kotlin/id/walt/commons/exceptions/CryptoExceptions.kt

```diff
@@ -0,0 +1,56 @@
+package id.walt.commons.exceptions
+
+import io.ktor.http.HttpStatusCode
+
+open class CryptoException(val status: HttpStatusCode, message: String) : Exception(message)
+
+
+class KeyTypeNotSupportedException(type: String) :
+    CryptoException(HttpStatusCode.BadRequest, "Key type $type not supported")
+
+class KeyBackendNotSupportedException(backend: String) :
+    CryptoException(HttpStatusCode.BadRequest, "Key backend $backend not supported")
+
+class KeyTypeMissingException() :
+    CryptoException(HttpStatusCode.BadRequest, "Key type is missing in the serialized key")
+
+class KeyNotFoundException(
+    id: String,
+    message: String = "Key with id $id not found"
+) : CryptoException(HttpStatusCode.NotFound, message)
+
+class SigningException(message: String) : CryptoException(HttpStatusCode.InternalServerError, message)
+class VerificationException(message: String) : CryptoException(HttpStatusCode.InternalServerError, message)
+
+class MissingSignatureException(message: String) : CryptoException(HttpStatusCode.BadRequest, message)
+object TSEError {
+    class LoginException(errors: List<String>) : CryptoException(
+        HttpStatusCode.InternalServerError,
+        "Errors occurred at TSE login: ${errors.joinToString { it }}"
+    )
+
+    class MissingAuthTokenException :
+        CryptoException(HttpStatusCode.InternalServerError, "Did not receive token after login!")
+
+    class MissingKeyNameException : CryptoException(HttpStatusCode.BadRequest, "Key name is missing in the request!")
```

Comment:
nitpicking alert!
I think the correct form in this case would be "missing from"